cpp/tests/test_io_cli.cpp

```diff
@@ -17,21 +17,70 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-#include "memilio/config.h"
+#include "matchers.h"
 #include "memilio/io/cli.h"
+#include "memilio/io/io.h"
+#include "memilio/io/json_serializer.h"
 #include "temp_file_register.h"
-#include "utils.h"
 
 #ifdef MEMILIO_HAS_JSONCPP
 
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
 
+#include <memory>
 #include <sstream>
 #include <string>
 #include <vector>
 #include <list>
 
+namespace
+{
+
+class TestCLI : public ::testing::Test
+{
+public:
+    // allocates a char** argv, filled with the given values. arrays are freed during TearDown
+    std::pair<int, char**> make_argv(const std::vector<std::string>& values)
+    {
+        const int argc = static_cast<int>(values.size());
+        char** argv    = new char*[argc];
+        // create and copy each string
+        for (int i = 0; i < argc; i++) {
+            const auto n = values[i].size() + 1;
+            argv[i]      = new char[n];
+            for (size_t j = 0; j < n; j++) {
```

Comment:
the last index is out of bounds, right? Because j goes to n (and n is size()+1)

Reply:
Nope, welcome to string handling in C... That +1 is the terminating null char, without this, stuff will break.